eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php

```diff
@@ -0,0 +1,416 @@
+        $sectionId = $this->getFirstSearchFieldValue($document->fields, 'section_id');
+        $published = $this->getFirstSearchFieldValue($document->fields, 'publication_date');
+
+        for ($i = 0; $i < count($indexArray); ++$i) {
+            $indexWord = $indexArray[$i]['Word'];
+            $contentFieldId = $indexArray[$i]['ContentClassAttributeID'];
+            $identifier = $indexArray[$i]['identifier'];
+            $integerValue = $indexArray[$i]['integer_value'];
+            $wordId = $wordIDArray[$indexWord];
+
+            if (isset($indexArray[$i + 1])) {
+                $nextIndexWord = $indexArray[$i + 1]['Word'];
+                $nextWordId = $wordIDArray[$nextIndexWord];
+            } else {
+                $nextWordId = 0;
+            }
+            $frequency = 0;
+            $this->searchIndex->addObjectWordLink($wordId, $contentId, $frequency, $placement, $nextWordId, $prevWordId, $contentTypeId, $contentFieldId, $published, $sectionId, $identifier, $integerValue);
+            $prevWordId = $wordId;
+            ++$placement;
+        }
+
+        return $placement;
+    }
+
+    /**
+     * Find first \eZ\Publish\SPI\Search\Field with given name and return its value.
+     *
+     * @param \eZ\Publish\SPI\Search\Field[] $fields
+     * @param string $name
+     * @return mixed
+     */
+    private function getFirstSearchFieldValue(array $fields, $name)
+    {
+        $filtered = array_values(
+            array_filter($fields, function (Field $field) use ($name) {
+                return $field->name === $name;
+            })
+        );
+
+        return isset($filtered[0]) ? $filtered[0]->value : null;
+    }
+
+    /**
+     * Remove duplicated spaces.
+     *
+     * @param string $text
+     *
+     * @return string
+     */
+    private static function removeDuplicatedSpaces($text)
+    {
+        return preg_replace('/\s{2,}/', ' ', $text);
+    }
+
+    /**
+     * Remove single and double quotes, including UTF-8 variations.
+     *
+     * @param string $text
+     *
+     * @return string
+     */
+    private static function removeAllQuotes($text)
+    {
+        return preg_replace('/([\x{2018}-\x{201f}]|\'|")/u', ' ', $text);
+    }
+
+    /**
+     * Build WordIDArray and update ezsearch_word table.
+     *
+     * Ported from the legacy code
+     *
+     * @see https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/search/plugins/ezsearchengine/ezsearchengine.php#L155
+     *
+     * @param array $indexArrayOnlyWords words for object to add
+     *
+     * @return array wordIDArray
+     */
+    private function buildWordIDArray(array $indexArrayOnlyWords)
+    {
+        $wordCount = count($indexArrayOnlyWords);
+        $wordIDArray = array();
+        $wordArray = array();
+        // store the words in the index and remember the ID
+        $dbName = $this->dbHandler->getName();
+
+        if ($dbName == 'mysql') {
```

Comment:
Yes we can :)
[Postgresql doesn't cast words which are integers](https://travis-ci.org/ezsystems/ezpublish-kernel/jobs/139475379#L630), so I fixed it like this:
`->where($query->expr->in('word', array_map('strval', $words)));`
and it works now without those two `if`s